product-service/src/lambdas/catalogBatchProcess.ts

```diff
@@ -0,0 +1,62 @@
+import { APIGatewayEventRequestContext } from "aws-lambda";
+import cors from "../../data/cors.json";
+import { Client, ClientConfig } from "pg";
+import dbOptions from "../../data/db";
+import Product from "../models/Product";
+import AWS from "aws-sdk";
+
+const createProduct =
+  "INSERT INTO products (title, description, price) VALUES ($1, $2, $3) RETURNING *";
+
+const createStock =
+  "INSERT INTO stocks (product_id, count) VALUES ($1, $2) RETURNING count";
+
+export const add = async (
+  event,
+  context: APIGatewayEventRequestContext,
+  callback
+): Promise<void> => {
+  let client;
+  try {
+    console.log(event);
+    const products = event.Records.map(
+      ({ body }) => JSON.parse(body) as Product
+    ) as Product[];
+    console.log(products);
+    const sns = new AWS.SNS({ region: "eu-west-1" });
+    client = new Client((dbOptions as unknown) as ClientConfig);
```

Comment:
The main idea for this lambda was to post a request with product data to the 'createProduct' lambda and then that lambda should write it to DB.

Reply:
As far as I understood - it should do it by itself.
"Configure the SQS to trigger lambda catalogBatchProcess with 5 messages at once via batchSize property. The lambda function should iterate over all SQS messages **and create corresponding products in the products table**."

Reply:
You might be right. There is some confusion between requirements, QA session, and QA document.